### DISCUSSION TITLE
Trying to attach an image in public folder with nodemailer fails in Vercel but works in local in dev and production build

### DISCUSSION THREAD

**Discussion by albertnev** [vercel/next.js#67698] [Help] [answered]
### Summary

I'm using `nodemailer` to send some e-mails when a contact form is submitted. It was working perfectly, until I tried to send an image inside the HTML body. After some investigation, I found that I had to add an `attachments` prop to the `Mail.Options` object, and for it to work with NextJS I had to use `path.join()` and `process.cwd()` to create the path, so I did so:

```
"use server";

export const sendFormEmail = async (data: Record<string, string>) => {
    ...
    const emailImagePath = path.join(process.cwd(), "public", "email_image.png");
    const mailOptions: Mail.Options = {
        attachments: [
          {
            cid: "emailimage@uniqueid.com", // just a unique ID string to be used like <img src="cid:emailimage@..." />
            filename: "email_image.png",
            path: emailImagePath,
          },
        ],
        ...
    };
    ...
    // try { ... } catch (error) { ... }
    await transport.sendMail(mailOptions);
    ...
};
```
**And it works. In local, at least.** It doesn't matter if it's in _dev_ (`npm run dev`) or _production_ (`npm run start`) environment, it will send the e-mail with the image displayed in its body correctly. However, when I deploy it to Vercel, the e-mailing process throws with the following error:

```
{
    "errno": -2,
    "code": "ESTREAM",
    "syscall": "open",
    "path": "/var/task/public/email_image.png",
    "command": "API"
}
```
The image is located inside my project root `/public/` folder, at `/public/email_image.png`. 

What am I doing wrong? How do I have to create the image path to be able to use `nodemailer` to open it and attach it into the e-mail body?

Thank you. 

### Additional information

```js
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Pro
  Available memory (MB): 32550
  Available CPU cores: 8
Binaries:
  Node: 20.11.1
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.2.3
  eslint-config-next: 14.2.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.5.3
Next.js Config:
  output: N/A
-----
My package dependencies: 

"next": "14.2.3",
"nodemailer": "^6.9.14",
"react": "18.2.0",
"react-dom": "18.2.0",
"react-email": "2.1.5",
```

**Comment by Tomin-Joy** [comment]
Facing the same issue now. Did you found any solution

**Reply by albertnev** [reply, marked as answer]
Yes, although it's not a solution per se! Basically, I just published the code with the image in the public folder into production, and used the `VERCEL_PROJECT_PRODUCTION_URL` automatic environment variable to do so (this piece of code is in a `use server` function, so it has access to private environment variables):

```
  const emailImagePath = process.env.VERCEL_PROJECT_PRODUCTION_URL
      ? `https://${process.env.VERCEL_PROJECT_PRODUCTION_URL}/email_image.png`
      : path.join(process.cwd(), "public", "email_image.png");

  const mailOptions: Mail.Options = {
        attachments: [
          {
            cid: "emailimage@uniqueid.com", 
            filename: "email_image.png",
            path: emailImagePath,
          },
        ],
        ...
    };
```
This is not a solution, as you're not really using the image in your codebase, but the one published in production. This means if you change your image to a new one, you won't see it until you publish your new image into the production site, but at least it's a workaround that works.
